hack/update-bazel.sh

```diff
@@ -17,6 +17,10 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
+if ! which -s bazel && [[ "${CALLED_FROM_MAIN_MAKEFILE:-}" == "" ]]; then
```

Comment:
explain this?  It is not true for most hack scripts.

Reply:
If we are going to delete the generated code we need to generate the files in order to calculate them. 

Thus the user either needs to have bazel installed (which will generate the files) or use the Makefile rule to run updates (which adds a dep on the generate rule)